### PR TITLE
CRM-16517 add OpenID Api methods

### PR DESCRIPTION
**Pull request by mallezie**
I've given this a first start. When i try to run the added tests, this fails lacally on the create statements, but i'm not sure what causes this, so i'm gonnan test it here first.

### CODE REVIEW

File: tests/phpunit/api/v3/OpenIDTest.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      'name' => 'My OpenID handle',
+      'location_type_id' => 1,
+    );
+  }
+
+  public function testCreateOpenID() {
+    $result = $this->callAPIAndDocument($this->_entity, 'create', $this->params, __FUNCTION__, __FILE__);
+    $this->assertEquals(1, $result['count'], 'In line ' . __LINE__);
+    $this->getAndCheck($this->params, $result['id'], $this->_entity);
+    $this->assertNotNull($result['values'][$result['id']]['id'], 'In line ' . __LINE__);
```
Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/OpenIDTest.php]: minor point but we don't need the 'in line ' __LINE__ anymore

**Comment by mallezie** [comment]
I'll probably need to check casing of the create and delete methods. Just posting here for reference, i'll update this PR tonight, or tomorrow. 
```
(09:11:01) eileen: I think the issue is that openid should probably be open_id
(09:11:14) eileen: The casing is a bit wierd
(09:11:28) eileen: for that entity
(09:11:49) eileen: but it will be looking for function civicrm_api3_open_id_create
(09:11:56) eileen: or possibly function civicrm_api3_open_i_d_create
...
(09:13:45) eileen: yes - probably in api.php it's figuring out the function name
```

File: tests/phpunit/api/v3/OpenIDTest.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  public function testDeleteOpenID() {
+    $result = $this->callAPISuccess($this->_entity, 'create', $this->params);
+    $deleteParams = array('id' => $result['id']);
+    $result = $this->callAPIAndDocument($this->_entity, 'delete', $deleteParams, __FUNCTION__, __FILE__);
+    $checkDeleted = $this->callAPISuccess($this->_entity, 'get', array());
+    $this->assertEquals(0, $checkDeleted['count'], 'In line ' . __LINE__);
+  }
+
+  public function testDeleteOpenIDInvalid() {
```
Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/OpenIDTest.php]: Another aside. I think these 'failure tests' are at best marginal. The wrapper itself is pretty heavily tested so unless you have a specific bit of code to test I don't see value in the invalid data tests

Comment (mallezie) [inline tests/phpunit/api/v3/OpenIDTest.php]: I'll check it out. I actually just copied the IM-tests.

Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/OpenIDTest.php]: Yeah - there are a lot of unnecessary tests around!

**Comment by mallezie** [comment]
This should be better, this makes the API useable through the explorer, and makes the tests pass. 
Not sure why the tests didn't fail the first time? Looks like they weren't exectuded by default? Do i need to do something extra to get them run by the testbot?

**Comment by mallezie** [comment]
Now I see the test executed by the bot. So that's good ;-)

**Comment by eileenmcnaughton** [comment]
The failed test is because the schema does not have titles. The api enforces that all fields exposed through it must have a title

**Comment by mallezie** [comment]
I've added the titles, this should fix the tests. 
@eileenmcnaughton about the proposed test-changes, i've noticed the current tests here come forward in lot's of places. (EmailTest, IMTest, TagTest, ContactTypeTest, ...). They seem to be some sort of 'established' pattern. Although as you say probably wron / unnecessary. Shouldn't we better clean this up in a followup for all tests?

**Comment by eileenmcnaughton** [comment]
@mallezie Yes we *should* - I have been doing various tidy ups when I touch tests but I haven't pro-actively gone through them all

**Comment by mallezie** [comment]
@eileenmcnaughton i'm willing to help, let's keep it out of this PR for now, and we'll think on IRC over a cleanup plan?

**Comment by eileenmcnaughton** [comment]
Ah yes - it's GREEN isn't it ... merging!

**Pull request merged**